monkestation/code/modules/antagonists/mimic/mimic.dm

```diff
@@ -0,0 +1,561 @@
+	obj_damage = 30
+	see_in_dark = 8
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	wander = FALSE
+	initial_language_holder = /datum/language_holder/mimic
+	attacktext = "smothers"
+	attack_sound = 'monkestation/sound/creatures/mimic/mimicattack.ogg'
+	var/absorb_sound = 'monkestation/sound/creatures/mimic/mimicabsorb.ogg'
+	var/split_sound = 'monkestation/sound/creatures/mimic/mimicsplit.ogg'
+
+	var/playstyle_string = "<span class='big bold'>You are a mimic,</span></b> an alien that made it's way on to the station. \
+							You can take the form of any item you can see by clicking on it. You can latch onto people by clicking on them, \
+							which is instant when you're disguised. When you latch onto someone, they can't hurt you, but other people \
+							can. After someone dies, you can absorb their body and reproduce to make more mimics.</b>"
+
+	var/disguised = FALSE
+	var/atom/movable/form = null
+	var/disguise_time = 0
+	var/static/people_absorbed = 0
+
+	//The number of current mimics
+	var/static/mimic_count = 0
+	//The name used in the hivemind chat
+	var/hivemind_name
+
+	//If this is currently reproducing
+	var/splitting = FALSE
+	//The target npc mimic's try to disguise as.
+	var/atom/movable/ai_disg_target = null
+	//attempts to reach a disguise target
+	var/ai_disg_reach_attempts = 0
+
+	var/fleeing = FALSE
+	mobchatspan = "blob"
+	discovery_points = 2000
+
+/mob/living/simple_animal/hostile/alien_mimic/examine(mob/user)
+	. = ..()
+	if(disguised)
+		. += "<span class='warning'>It jitters a little bit...</span>"
+
+/mob/living/simple_animal/hostile/alien_mimic/Destroy()
+	mimic_count--
+	ai_disg_target = null
+	. = ..()
+
+/mob/living/simple_animal/hostile/alien_mimic/med_hud_set_health()
+	if(disguised && !isliving(form))
+		var/image/holder = hud_list[HEALTH_HUD]
+		holder.icon_state = null
+		return //we hide medical hud while disguised
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/med_hud_set_status()
+	if(disguised && !isliving(form))
+		var/image/holder = hud_list[STATUS_HUD]
+		holder.icon_state = null
+		return //we hide medical hud while disguised
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/ClickOn(atom/target_item)
+	if(allowed(target_item)) //Become Items
+		if(disguise_time > world.time)
+			to_chat(src, "<span class='warning'>You diguised too recently, wait a little longer!</span>")
+			return
+		if(disguised)
+			restore()
+		var/obj/item/item = target_item
+		if(!item.anchored)
+			disguise(item)
+			return
+	. = ..()
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/allowed(atom/movable/target_item)
+	return isitem(target_item) && !istype(target_item, /obj/item/radio/intercom)
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/is_table(atom/possible_table)
+	return istype(possible_table, /obj/structure/table) || istype(possible_table, /obj/structure/rack)
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/replication_cost()
+	return 1 + round((mimic_count / 2))
```

Comment:
Unneeded proc, just put it right into where it needs to be OR create a macro for it.